crates/red_knot_python_semantic/src/types.rs

```diff
@@ -489,7 +490,9 @@ pub enum Type<'db> {
     SubclassOf(SubclassOfType<'db>),
     /// The set of Python objects with the given class in their __class__'s method resolution order.
     /// Construct this variant using the `Type::instance` constructor function.
-    Instance(InstanceType<'db>),
+    NominalInstance(NominalInstanceType<'db>),
```

Comment:
As a favor to reviewers, next time you might consider splitting out a mass rename like this into its own PR :)

Reply:
No worries, just a thought! Since you raised the question, I think the multi-PR sequencing that I was imagining is that when you had the idea that maybe we should rename `Instance` to `NominalInstance` if we add `ProtocolInstance`, there's still really no pressing need to do that rename right away. Having both `Instance` and `ProtocolInstance` is maybe confusing (though I think tbh it would probably be fine even if left that way) but otherwise not a problem. So it could be left to a later follow-up.